src/Database/LSMTree/Internal.hs

```diff
@@ -787,14 +787,14 @@ flushWriteBuffer conf@TableConfig{confDiskCachePolicy}
 -- | Note that the invariants rely on the fact that levelling is only used on
 -- the last level.
 --
-levelsInvariant :: forall s h. TableConfig -> Levels h -> ST s Bool
+levelsInvariant :: forall s1 s2 h. TableConfig -> Levels s2 h -> ST s1 Bool
```

Comment:
this doesnt look right, why different state threads?

Reply:
The current invariant check doesn't actually use any mutable references, but it will once we add merge scheduling, so I'll change it to account for that in this PR.